tests/sagemaker/mock/__init__.py

```diff
@@ -0,0 +1,629 @@
+        return response
+
+
+class EndpointConfigDescription:
+    """
+    Object representing an endpoint configuration description returned by SageMaker's
+    "DescribeEndpointConfig" API:
+    https://docs.aws.amazon.com/sagemaker/latest/dg/API_DescribeEndpointConfig.html.
+    """
+
+    def __init__(self, config, arn):
+        self.config = config
+        self.arn = arn
+
+    @property
+    def response_object(self):
+        response = {
+            'EndpointConfigName': self.config.config_name,
+            'EndpointConfigArn': self.arn,
+            'ProductionVariants': self.config.production_variants,
+            'CreationTime': self.config.creation_time,
+        }
+        return response
+
+
+class Model(TimestampedResource):
+    """
+    Object representing a SageMaker model. The SageMakerBackend will create and manage Models.
+    """
+
+    def __init__(self, model_name, primary_container, execution_role_arn, tags, vpc_config):
+        super(Model, self).__init__()
+        self.model_name = model_name
+        self.primary_container = primary_container
+        self.execution_role_arn = execution_role_arn
+        self.tags = tags
+        self.vpc_config = vpc_config
+
+    @property
+    def arn_descriptor(self):
+        return ":model/{model_name}".format(model_name=self.model_name)
+
+
+class ModelSummary:
+    """
+    Object representing a model entry in the models list returned by SageMaker's
+    "ListModels" API: https://docs.aws.amazon.com/sagemaker/latest/dg/API_ListModels.html.
+    """
+
+    def __init__(self, model, arn):
+        self.model = model
+        self.arn = arn
+
+    @property
+    def response_object(self):
+        response = {
+            'ModelArn': self.arn,
+            'ModelName': self.model.model_name,
+            'CreationTime': self.model.creation_time,
+        }
+        return response
+
+
+class ModelDescription:
+    """
+    Object representing a model description returned by SageMaker's
+    "DescribeModel" API: https://docs.aws.amazon.com/sagemaker/latest/dg/API_DescribeModel.html.
+    """
+
+    def __init__(self, model, arn):
+        self.model = model
+        self.arn = arn
+
+    @property
+    def response_object(self):
+        response = {
+            'ModelArn': self.arn,
+            'ModelName': self.model.model_name,
+            'PrimaryContainer': self.model.primary_container,
+            'ExecutionRoleArn': self.model.execution_role_arn,
+            'VpcConfig': self.model.vpc_config if self.model.vpc_config else {},
+            'CreationTime': self.model.creation_time,
+        }
+        return response
+
+
+# Create a SageMaker backend for each EC2 region
+sagemaker_backends = {}
+for region, ec2_backend in ec2_backends.items():
+    new_backend = SageMakerBackend()
+    sagemaker_backends[region] = new_backend
+
+mock_sagemaker = base_decorator(sagemaker_backends)
+mock_sagemaker_deprecated = deprecated_base_decorator(sagemaker_backends)
```

Comment:
Why do we need the deprecated_base_decorator? It sounds... deprecated.

Reply:
Including a "deprecated" decorator is a convention used throughout the various mocks that are included with Moto; this doesn't seem to be useful in our case, so I've removed it.